docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/migrate-front-end.md

```diff
@@ -151,6 +155,201 @@ export default {
+    },
+    // links
+    [
+      {
+        intlLabel: {
+          id: 'settings.page',
+          defaultMessage: 'Setting page 1',
+        },
+        id: 'settings',
+        to: `/settings/my-plugin/`,
+        Component: async () => {
+          const component = await import(
+            /* webpackChunkName: "my-plugin-settings-page" */ './pages/Settings'
+          );
+
+          return component;
+        },
+        permissions: [],
+      },
+
+    ]
+  );
+
+  app.registerPlugin({
+    id: pluginId,
+    name,
+  });
+},
+```
+
+:::
+
+## Adding reducers
+
+::: strapi v3/v4 comparison
+
+A Strapi v3 plugin adds reducers by exporting a `reducers` property during the plugin registration.
+
+In Strapi v4, a plugin adds reducers programmatically using the [Reducers API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#reducers-api).
+:::
+
+To migrate to Strapi v4, make sure reducers are added programmatically with the `addReducers()` method.
+
+<!-- ? did the reducers format change in v4? since it's not documented in Strapi v3, I can't compare… 😅  ? -->
+
+::: details Example of adding reducers
+
+```js
+// path: ./src/plugins/my-plugin/admin/src/index.js
+
+import myReducer from './components/MyCompo/reducer';
+import myReducer1 from './components/MyCompo1/reducer';
+import pluginId from './pluginId';
+
+const reducers = {
+  [`${pluginId}_reducer`]: myReducer,
+  [`${pluginId}_reducer1`]: myReducer1,
+};
+
+export default {
+  register(app) {
+    app.addReducers(reducers);
+
+    app.registerPlugin({
+      id: pluginId,
+      initializer: Initializer,
+      isReady: false,
+      name,
+    });
+  },
+ }
+}
+```
+
+:::
+
+## Adding injection zones
+
+::: strapi v3/v4 comparison
+A Strapi v3 plugin can inject components into the Content Manager's Edit view, using the `registerField()` method or the `useStrapi` hook within the `Initializer` component.
+
+In Strapi v4, a plugin can inject components into several locations of the Content Manager using the [Injection Zones API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#injection-zones-api).
+:::
+
+To migrate to Strapi v4, make sure components are injected using Strapi v4 [Injection Zones API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#injection-zones-api). Depending on where the component should be injected, use:
+
+- either the `injectContentManagerComponent()` method to inject into [predefined injection zones](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#using-predefined-injection-zones) of the Content Manager
+- or the `injectComponent()` method to inject into [custom zones](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#creating-a-custom-injection-zone)
+
+::: details Example of injecting a component into the Content Manager's Edit view
+
+```jsx
+// path:  ./src/plugins/my-plugin/admin/src/index.js
+
```

Comment:
Can you add the pluginId import as well?

`import pluginId from './pluginId`;` as it will be used to prefix the name of the injected component?